src/protocols/screencopy.rs

```diff
@@ -275,44 +351,50 @@ where
 
         let (buffer, with_damage) = match request {
             zwlr_screencopy_frame_v1::Request::Copy { buffer } => (buffer, false),
-            // zwlr_screencopy_frame_v1::Request::CopyWithDamage { buffer } => (buffer, true),
+            zwlr_screencopy_frame_v1::Request::CopyWithDamage { buffer } => (buffer, true),
             _ => unreachable!(),
         };
 
-        if !shm::with_buffer_contents(&buffer, |_buf, shm_len, buffer_data| {
```

Comment:
The size and format check should remain here because this is the only place where we can clearly check for a protocol violation (we asked for a buffer of specific format and size, and the client sent something different). If subsequent rendering finds that the buffer size (that we asked for) is now wrong, that should produce a failed event.

Reply:
The rendering is checking the buffer size and format against specification though. I want to avoid double checking if possible.

Reply:
I don't think you can avoid the double size checking since the output can resize at any time (which shouldn't kill the screencopy client with a protocol error). It's not a big deal to check it twice